### PR TITLE
[Spike] Display `unsupported` dialog message for Firefox & Safari users

### LINKED ISSUE #46: Display an unsupported dialog for Firefox and Safari
"Sorry, this browser is not supported by this app. To participate in our Governance Platform, please use Chrome or its variants on a Mac or PC."

### PR DESCRIPTION
**Pull request by aidan-starke**
## Summary
 - Closes #46
 - Prevent `promptInstallExtension` confirm in favour of `UserAgentDialog` for Firefox & Safari browser
 - Sneak in copy update for `social channels` section
 
 ## Notes
  - I had to allow for an `undefined` state as useEffect was picking up the default state of `ReadWrite` too quickly on Safari

### CODE REVIEW

File: libs/web/providers/src/CENNZExtensionProvider.tsx [hunk 1]
```diff
@@ -40,20 +40,14 @@ export const CENNZExtensionProvider: FC<CENNZExtensionProviderProps> = ({
...
 	const promptInstallExtension = useCallback(() => {
-		if (
-			browser.name === "Safari" ||
-			os.name === "iOS" ||
-			os.name === "Android"
-		) {
+		if (os.name === "iOS" || os.name === "Android") {
 			return alert(
-				"Sorry, this browser is not supported by this app. To use this app, please switch to Chrome or Firefox browsers on a Mac or PC."
+				"Sorry, this browser is not supported by this app. To use this app, please switch to Chrome or its on a Mac or PC."
```
Comment (ken-futureverse) [inline libs/web/providers/src/CENNZExtensionProvider.tsx]:
```suggestion
				"Sorry, this browser is not supported by this app. To use this app, please switch to Chrome or its variants on a Mac or PC."
```

File: libs/web/providers/src/UserAgentProvider.tsx [hunk 2]
```diff
@@ -36,13 +45,40 @@ export const UserAgentProvider: FC<UserAgentProviderProps> = ({
...
 	return (
-		<UserAgentContext.Provider value={userAgent}>
-			{children}
-		</UserAgentContext.Provider>
+		<div>
+			<UserAgentContext.Provider value={userAgent}>
+				{children}
+			</UserAgentContext.Provider>
+
+			<UserAgentDialog open={open} onClose={() => setOpen(false)} />
```
Comment (ken-futureverse) [inline libs/web/providers/src/UserAgentProvider.tsx]: should be `BrowsersDialog`

**Review by ken-futureverse — CHANGES_REQUESTED** [review]
Should have this in the UserAgentProvider so we don't have to scatter the browser detection logic everywhere

```
type RuntimeMode = "ReadOnly" | "ReadWrite";
...
const [runtimeMode, setRuntimeMode] = useState<RuntimeMode>("ReadWrite");

useEffect(() => {
	if (!browser?.name || !os?.name) return;
	if (browser.name === "Safari" || os.name === "iOS" || os.name === "Android")
		setRuntimeMode("ReadOnly");
}, [browser?.name, os?.name]);
```

File: libs/web/providers/src/CENNZExtensionProvider.tsx (hunk 1 above)
Comment (aidan-starke) [reply libs/web/providers/src/CENNZExtensionProvider.tsx]: removed as this code won't be reached

File: libs/web/providers/src/CENNZExtensionProvider.tsx [hunk 3]
```diff
@@ -73,19 +67,15 @@ export const CENNZExtensionProvider: FC<CENNZExtensionProviderProps> = ({
 	}, [appName, module]);
 
 	useEffect(() => {
-		if (!module || !browser?.name) return;
+		if (!module || !runtimeMode || runtimeMode === "ReadOnly") return;
```
Comment (ken-futureverse) [inline libs/web/providers/src/CENNZExtensionProvider.tsx:70]: just a question here, why do need `!runtimeMode` here?

**Review by ken-futureverse — APPROVED** [review]

File: libs/web/providers/src/CENNZExtensionProvider.tsx (hunk 3 above)
Comment (aidan-starke) [reply libs/web/providers/src/CENNZExtensionProvider.tsx:70]: useEffect was picking up the default state of `ReadWrite` too quickly on Safari

**Pull request merged**